### PR TITLE
[EDX-444] Support for server-side redirects with nginx

### PR DESCRIPTION
**Pull request by kennethkalmer**
## Description

Support for server-side redirects via nginx, meaning we can send proper `301` redirects instead of rendering pages and doing client-side redirects.

This can be achived by generating a map of redirects in `config/nginx-redirects.conf`, which nginx will load up and use.

* [EDX-444](https://ably.atlassian.net/browse/EDX-444)

## Review

Instructions on how to review the PR. 

* [Page to review](link)

### CODE REVIEW

File: config/nginx.conf.erb [hunk 1]
```diff
@@ -32,6 +32,14 @@ http {
 	# Must read the body in 5 seconds.
 	client_body_timeout <%= ENV['NGINX_CLIENT_BODY_TIMEOUT'] || 5 %>;
 
+  # Creates a map of redirects for us
+  map $uri $redirected_url {
+    default "none";
+    <% if File.exists?(File.dirname(__FILE__) + '/nginx-redirectss.conf') -%>
```
Comment (MymmiJ) [inline config/nginx.conf.erb:38]: minor typo! too many s's

**Review by MymmiJ — APPROVED** [review]
Minor typo in config/nginx.conf.erb otherwise we could get this portion merged now, and I'll write out to the nginx-redirects.conf file in a separate PR?

**Comment by kennethkalmer** [comment]
> Minor typo in config/nginx.conf.erb otherwise we could get this portion merged now, and I'll write out to the nginx-redirects.conf file in a separate PR?

@MymmiJ let's use a single PR to get this done :)

**Comment by kennethkalmer** [comment]
@MymmiJ completely reworked the history to be nice and tidy ✨

**Review by MymmiJ — APPROVED** [review]
LGTM!

**Pull request merged**